syntax = "proto3";

package protobufs;

message ParamsRequest {
  bytes para_request = 1;  
}

// normalized training data
message ParamsResponse {  
  bytes para_response = 2;
}

// service method
service MonaiFLService{
  rpc ParamTransfer(ParamsRequest) returns (ParamsResponse);
}


// syntax = "proto3";

// service MonaiService {
//   rpc Join(stream ClientMessage) returns (stream ServerMessage) {}
// }

// message Parameters {
//   repeated bytes tensors = 1;
//   string tensor_type = 2;
// }
// enum Reason {
//   UNKNOWN = 0;
//   RECONNECT = 1;
//   POWER_DISCONNECTED = 2;
//   WIFI_UNAVAILABLE = 3;
//   ACK = 4;
// }

// message ServerMessage {
//   message Reconnect { int64 seconds = 1; }
//   message GetParameters {}
//   message FitIns {
//     Parameters parameters = 1;
//     map<string, Scalar> config = 2;
//   }
//   message EvaluateIns {
//     Parameters parameters = 1;
//     map<string, Scalar> config = 2;
//   }

//   oneof msg {
//     Reconnect reconnect = 1;
//     GetParameters get_parameters = 2;
//     FitIns fit_ins = 3;
//     EvaluateIns evaluate_ins = 4;
//   }
// }

// message ClientMessage {
//   message Disconnect { Reason reason = 1; }
//   message ParametersRes { Parameters parameters = 1; }
//   message FitRes {
//     Parameters parameters = 1;
//     int64 num_examples = 2;
//     int64 num_examples_ceil = 3 [ deprecated = true ];
//     float fit_duration = 4 [ deprecated = true ];
//     map<string, Scalar> metrics = 5;
//   }
//   message EvaluateRes {
//     int64 num_examples = 1;
//     float loss = 2;
//     float accuracy = 3 [ deprecated = true ];
//     map<string, Scalar> metrics = 4;
//   }

//   oneof msg {
//     Disconnect disconnect = 1;
//     ParametersRes parameters_res = 2;
//     FitRes fit_res = 3;
//     EvaluateRes evaluate_res = 4;
//   }
// }

// message Scalar {
//   // The following oneof contains all types that ProtoBuf considers to be
//   // "Scalar Value Types". Commented-out types are listed for reference and
//   // might be enabled in future releases. Source:
//   // https://developers.google.com/protocol-buffers/docs/proto3#scalar
//   oneof scalar {
//     double double = 1;
//     // float float = 2;
//     // int32 int32 = 3;
//     // int64 int64 = 4;
//     // uint32 uint32 = 5;
//     // uint64 uint64 = 6;
//     // sint32 sint32 = 7;
//     sint64 sint64 = 8;
//     // fixed32 fixed32 = 9;
//     // fixed64 fixed64 = 10;
//     // sfixed32 sfixed32 = 11;
//     // sfixed64 sfixed64 = 12;
//     bool bool = 13;
//     string string = 14;
//     bytes bytes = 15;
//   }
// }